### PR TITLE
[CodeQuality] Add Variable support on class arg on InlineIsAInstanceOfRector

### PR DESCRIPTION
**Pull request by samsonasik**
Per @staabm comment on PR:

- https://github.com/rectorphp/rector-src/pull/3258#discussion_r1059655585

This PR add `Variable` support on transforming `is_a` to `instanceof` on `InlineIsAInstanceOfRector`, ref https://3v4l.org/WUiel

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed :tada: @TomasVotruba it is ready for review.

File: rules-tests/CodeQuality/Rector/FuncCall/InlineIsAInstanceOfRector/Fixture/variable_class.php.inc [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+namespace Rector\Tests\CodeQuality\Rector\FuncCall\InlineIsAInstanceOfRector\Fixture;
+
+class VariableClass
+{
+    /**
+     * @param class-string<\PHPStan\Type\Type> $type
+     */
+    public function run(object $object, string $type)
+    {
+        return is_a($object, $type);
```
Comment (staabm) [inline rules-tests/CodeQuality/Rector/FuncCall/InlineIsAInstanceOfRector/Fixture/variable_class.php.inc:12]: I think you need another test which verifies that is_a() with a non-class-string 2nd arg is skipped

Comment (samsonasik) [reply rules-tests/CodeQuality/Rector/FuncCall/InlineIsAInstanceOfRector/Fixture/variable_class.php.inc:12]: That's not needed, all variables seems fine

Comment (samsonasik) [reply rules-tests/CodeQuality/Rector/FuncCall/InlineIsAInstanceOfRector/Fixture/variable_class.php.inc:12]: If there is a bug on variable usage for 2nd arg (class arg) after transformation, then the bug is on the original is_a

**Comment by TomasVotruba** [comment]
Very nice improvement :) thanks!

**Pull request merged**